.travis.yml

```diff
@@ -59,6 +63,7 @@ matrix:
       env: TARGET=go
 
 before_install:
+  - ./.travis/before-install-mvn3.3.9-trusty.sh
```

Comment:
@ericvergnaud exactly, have a look of this build:
https://travis-ci.org/xied75/antlr4/jobs/194889549 (line 2254)

The reasoning:

1, dotnet tooling has a minimum os requirement for ubuntu, which is Trusty.
2, If I switch Travis build to Trusty, it will be global for every targets
3, then you got this error
```
[ERROR] Failed to execute goal io.takari.maven.plugins:takari-lifecycle-plugin:1.11.12:testProperties (testProperties) on project antlr4-maven-plugin: The plugin io.takari.maven.plugins:takari-lifecycle-plugin:1.11.12 requires Maven version [3.2.1,) -> [Help 1]
```
4, Travis Trusty has the default mvn version 3.0.5, funny enough Travis bothered to update their Precise mvn to 3.2.5, and they refuse to do something similar for Trusty:
https://github.com/travis-ci/travis-ci/issues/4872
5, so that means if we choose Trusty, install a new version of mvn **globally** is a must, unless you are willing to lower/review the mvn plugin required minimum version.

Hope this clear?

(p.s. I realized changing to Trusty break the cpp and swift builds, which I can fix if you decide to go ahead with migrating to Trusty)